pkg/reconciler/gc/v2/gc.go

```diff
@@ -100,14 +102,18 @@ func Collect(
 	return nil
 }
 
+// nonactiveRevisions swaps active revisions to the end and omits them in new slice
```

Comment:
You're not returning a `new` slice per se :)
https://play.golang.org/p/rshnGeGCGPp